docs/src/manual/complex.md

```diff
@@ -251,3 +251,22 @@ GenericAffExpr{ComplexF64, VariableRef}
 julia> typeof(H[2, 1])
 GenericAffExpr{ComplexF64, VariableRef}
 ```
+
+## Hermitian PSD constraints
+
+The [`HermitianPSDCone`](@ref) can also be used in the [`@constraint`](@ref)
+macro:
+```jldoctest
+julia> model = Model();
+
+julia> @variable(model, H[1:2, 1:2], Symmetric)
+2×2 LinearAlgebra.Symmetric{VariableRef, Matrix{VariableRef}}:
+ H[1,1]  H[1,2]
+ H[1,2]  H[2,2]
+
+julia> import LinearAlgebra
+
+julia> @constraint(model, LinearAlgebra.Hermitian(H) in HermitianPSDCone())
+[H[1,1]                (1.0 + 1.0im) H[1,2];
```

Comment:
It's weird we get 1 + 1im and not 1 + 0im, looks like a bug

Reply:
I thought I was going crazy trying to debug this. I was like, where does the `1.0im` come from!?!?! Looks like it might be a printing bug:
```Julia
julia> model = Model()
A JuMP Model
Feasibility problem with:
Variables: 0
Model mode: AUTOMATIC
CachingOptimizer state: NO_OPTIMIZER
Solver name: No optimizer attached.

julia> @variable(model, x[1:2, 1:2])
2×2 Matrix{VariableRef}:
 x[1,1]  x[1,2]
 x[2,1]  x[2,2]

julia> import LinearAlgebra

julia> H = LinearAlgebra.Hermitian(x)
2×2 LinearAlgebra.Hermitian{VariableRef, Matrix{VariableRef}}:
 x[1,1]  x[1,2]
 x[1,2]  x[2,2]

julia> @constraint(model, H in HermitianPSDCone())
[x[1,1]                (1.0 + 1.0im) x[1,2];
 (1.0 - 1.0im) x[1,2]  x[2,2]] ∈ HermitianPSDCone()

julia> print(backend(model))
Feasibility

Subject to:

VectorOfVariables-in-HermitianPositiveSemidefiniteConeTriangle
 ┌      ┐
 │x[1,1]│
 │x[1,2]│
 │x[2,2]│
 │x[1,2]│
 └      ┘ ∈ HermitianPositiveSemidefiniteConeTriangle(2)
```

Reply:
The issue is here: https://github.com/jump-dev/JuMP.jl/blob/34462e415b50a628bd29dbf88eed60a38368f246/src/sd.jl#L482-L499

I think I need you to look at this; I'm not sure exactly what you intend to happen.

```julia
julia> model = Model()
A JuMP Model
Feasibility problem with:
Variables: 0
Model mode: AUTOMATIC
CachingOptimizer state: NO_OPTIMIZER
Solver name: No optimizer attached.

julia> @variable(model, x[1:2, 1:2])
2×2 Matrix{VariableRef}:
 x[1,1]  x[1,2]
 x[2,1]  x[2,2]

julia> import LinearAlgebra

julia> H = LinearAlgebra.Hermitian(x)
2×2 LinearAlgebra.Hermitian{VariableRef, Matrix{VariableRef}}:
 x[1,1]  x[1,2]
 x[1,2]  x[2,2]

julia> con = @constraint(model, H in HermitianPSDCone())
[x[1,1]                (1.0 + 1.0im) x[1,2];
 (1.0 - 1.0im) x[1,2]  x[2,2]] ∈ HermitianPSDCone()

julia> v = [x[1, 1], x[1, 2], x[2, 2], x[1, 2]]
4-element Vector{VariableRef}:
 x[1,1]
 x[1,2]
 x[2,2]
 x[1,2]

julia> reshape_vector(v, HermitianMatrixShape(2))
2×2 Matrix{GenericAffExpr{ComplexF64, VariableRef}}:
 x[1,1]                (1.0 + 1.0im) x[1,2]
 (1.0 - 1.0im) x[1,2]  x[2,2]
```

Reply:
I found the culprit. Fix incoming. It's a doozy